website/payments/admin_views.py

```diff
@@ -18,27 +20,41 @@
 )
 class PaymentAdminView(View):
     """
-    View that processes a payment
+    View that creates a payment
     """
 
     def post(self, request, *args, **kwargs):
-        payment = Payment.objects.filter(pk=kwargs["pk"])
+        if not (
+            kwargs.keys() >= {"app_label", "model_name", "payable"}
+            and "type" in request.POST
+        ):
+            raise SuspiciousOperation("Missing POST parameters")
+
+        if "next" in request.POST and not url_has_allowed_host_and_scheme(
+            request.POST.get("next"), allowed_hosts={request.get_host()}
+        ):
+            raise DisallowedRedirect
+
+        app_label = kwargs["app_label"]
+        model_name = kwargs["model_name"]
```

Comment:
Same thing as previous suggestions, why are these dict accesses instead of actual keyword arguments?